src/a11y-dialog.ts

```diff
@@ -192,20 +192,19 @@ export default class A11yDialog {
     // target
     // See: https://github.com/KittyGiraudel/a11y-dialog/issues/582
     const target = event.composedPath()[0] as HTMLElement
-
-    // We use `.closest(..)` and not `.matches(..)` here so that clicking
-    // an element nested within a dialog opener does cause the dialog to open
-    if (target.closest(`[${SCOPE}-show="${this.id}"]`)) {
-      this.show(event)
-    }
-
-    if (
-      target.closest(`[${SCOPE}-hide="${this.id}"]`) ||
-      (target.closest(`[${SCOPE}-hide]`) &&
-        target.closest('[aria-modal="true"]') === this.$el)
-    ) {
-      this.hide(event)
-    }
+    const opener = closest(`[${SCOPE}-show="${this.id}"]`, target)
```

Comment:
The code was massaged a bit to be easier to read but the fix is the use of `closest(..)`.